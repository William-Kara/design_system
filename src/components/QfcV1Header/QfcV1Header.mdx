import { ArgTypes, Meta } from '@storybook/blocks';
import QfcV1Header from '@/components/QfcV1Header/QfcV1Header.vue';

<Meta title="Modules/Headers/Header(v1)" />

# Definition & Usage

QfcHeader is placed at the top of the page. It contains a logo, includes the UserMenu component, and the button to
open the sidebar.

Please note that the QfcCard module is designed to be used in conjunction with the QfcSidebar component to create a complete user interface. The QfcSidebar provides a navigation menu that can be used to navigate to different sections of your application,

## Props

The following props are available for the QfcV1Header component:

<ArgTypes of={QfcV1Header} />

# Usage

To use QfcV1Header, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcV1Header.scss';
```

```html
<template>
  <QfcV1Header
    v-model:is-sidebar-open="isSidebarOpen"
    :user-menu="userMenu"
    :notification-button="notificationButton"
  />
</template>

<script>
  const isSidebarOpen = ref(false);
  const userMenu = {
    name: 'John Doe',
    links: [
      { link: '/#', title: 'Paramètre du compte' },
      { link: '/#', title: 'Profil' },
      { link: '/#', title: 'Déconnexion' },
    ],
    avatar: '/avatar.svg',
  };
</script>
```

# Examples

QfcV1Header should be used in conjunction with the QfcSidebar component to create a complete user interface.

```html
<QfcSidebar :items="sidebarItems" screen-id="main-container" v-model:is-open="isOpen" />
<main id="main-container">
  <QfcV1Header
    v-model:sidebar-is-open="isOpen"
    :user-menu="userMenu"
    :notification-button="notificationButton"
  />
</main>
```
