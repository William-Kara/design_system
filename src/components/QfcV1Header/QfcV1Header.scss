.qfc-header {
  height: 6.25rem;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: $background;
  color: $white;
  padding: 0 $q-generic-space-8;

  &__left-container, &__nav {
    display: flex;
    align-items: center;
    gap: $q-generic-space-2;
  }

  &__left-container {
    &__burger-container {
      display: flex;
      align-items: center;
      padding-right: $q-generic-space-10;
      gap: $q-generic-space-2;
      font-size: $fs-xs;
      background-color: transparent;
      color: $white;
      border: none;
      cursor: pointer;
    }
    &__logo {
      display: flex;
      font-size: $fs-m-2;
      align-items: center;
    }
  }
}

.fade-move,
.fade-enter-active,
.fade-leave-active {
  transition: all 0.4s ease, opacity 0.1s;
}

.fade-enter-from,
.fade-leave-to {
  opacity: 0;
  transform: translateX(-100%) !important;
}

.fade-enter-from {
  transform: translateX(-100%) translateY(0%) !important;
}

.fade-leave-active {
  position: absolute;
}
