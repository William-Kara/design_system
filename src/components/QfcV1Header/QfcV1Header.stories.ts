import V1Header from './QfcV1Header.vue';
import type { Meta, StoryObj } from '@storybook/vue3';
import Sidebar from '../QfcSidebar/QfcSidebar.vue';
import { Primary as SidebarPrimary } from '../QfcSidebar/QfcSidebar.stories';

const meta: Meta<typeof V1Header> = {
  component: V1Header,
  title: 'Modules/Headers/Header(v1)',
  argTypes: {
    isSidebarOpen: {
      control: { type: 'boolean', required: false },
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    isSidebarOpen: false,
    userMenu: {
      name: 'John Doe',
      links: [
        { link: '/#', title: 'Paramètre du compte' },
        { link: '/#', title: 'Profil' },
        { link: '/#', title: 'Déconnexion' },
      ],
      avatar: '/avatar.svg',
    },
    notificationButton: {
      to: '/',
      isActive: true,
      isNotification: true,
    },
  },
};

export const HeaderAndSidebar: Story = {
  render: (args) => ({
    components: { V1Header, Sidebar },
    setup() {
      return { args };
    },
    template: '<V1Header v-bind="args"></V1Header><Sidebar v-bind="args"/>',
  }),
  parameters: {
    layout: 'fullscreen',
  },
  args: {
    ...Primary.args,
    ...SidebarPrimary.args,
    isOpen: false,
    isSidebarOpen: false,
  },
};
