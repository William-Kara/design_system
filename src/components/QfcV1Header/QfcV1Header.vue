<template>
  <header class="qfc-header">
    <div class="qfc-header__left-container">
      <TransitionGroup name="fade">
        <button
          class="qfc-header__left-container__burger-container"
          @click="toggleSidebar"
          v-show="!isSidebarOpen"
          key="burger"
        >
          <i class="material-icons-outlined"> menu </i> Menu
        </button>
        <router-link to="/" class="qfc-header__left-container__logo" key="logo">
          <img src="/image-q.svg" width="32" height="25" alt="logo" />
        </router-link>
      </TransitionGroup>
    </div>

    <nav class="qfc-header__nav">
      <QfcSearchBar />
      <QfcButtonAction icon="notifications" label="notifications" v-bind="notificationButton" />
      <QfcUserMenu v-bind="userMenu" />
    </nav>
  </header>
</template>

<script lang="ts" setup>
import QfcUserMenu from '@/components/QfcUserMenu/QfcUserMenu.vue';
import QfcButtonAction from '@/components/QfcButtonAction/QfcButtonAction.vue';
import QfcSearchBar from '@/components/QfcSearchBar/QfcSearchBar.vue';

type RelativePath = `/${string}`;
type Links = { link: RelativePath; title: string }[];
interface Props {
  isSidebarOpen: boolean;
  userMenu: { name: string; links: Links; avatar: string };
  notificationButton: { to: RelativePath; isActive: boolean; isNotification: boolean };
}
defineProps<Props>();

const emit = defineEmits(['update:isSidebarOpen']);
function toggleSidebar() {
  emit('update:isSidebarOpen', true);
}
</script>

<style lang="scss" scoped src="./QfcV1Header.scss" />
