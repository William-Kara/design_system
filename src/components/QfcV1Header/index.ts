import QfcV1Header from './QfcV1Header.vue';

export { QfcV1Header };
export default QfcV1Header;
