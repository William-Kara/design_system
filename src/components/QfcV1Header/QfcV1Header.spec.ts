import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcV1Header from './QfcV1Header.vue';

let wrapper: any = null;

const propsValues = {
  isSidebarOpen: false,
};

beforeEach(() => {
  wrapper = shallowMount(QfcV1Header, {
    propsData: propsValues,
  });
});

describe('QfcV1Header Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
