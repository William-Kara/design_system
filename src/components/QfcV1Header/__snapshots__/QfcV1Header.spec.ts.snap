// Vitest Snapshot v1

exports[`QfcV1Header Component > should render correctly 1`] = `
"<header data-v-acb3ac67=\\"\\" class=\\"qfc-header\\">
  <div data-v-acb3ac67=\\"\\" class=\\"qfc-header__left-container\\">
    <transition-group-stub data-v-acb3ac67=\\"\\" name=\\"fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\"><button data-v-acb3ac67=\\"\\" class=\\"qfc-header__left-container__burger-container\\"><i data-v-acb3ac67=\\"\\" class=\\"material-icons-outlined\\"> menu </i> Menu </button>
      <router-link data-v-acb3ac67=\\"\\" to=\\"/\\" class=\\"qfc-header__left-container__logo\\"><img data-v-acb3ac67=\\"\\" src=\\"/image-q.svg\\" width=\\"32\\" height=\\"25\\" alt=\\"logo\\"></router-link>
    </transition-group-stub>
  </div>
  <nav data-v-acb3ac67=\\"\\" class=\\"qfc-header__nav\\">
    <qfc-search-bar-stub data-v-acb3ac67=\\"\\"></qfc-search-bar-stub>
    <qfc-button-action-stub data-v-acb3ac67=\\"\\" icon=\\"notifications\\" label=\\"notifications\\" isnotification=\\"false\\" isactive=\\"false\\"></qfc-button-action-stub>
    <qfc-user-menu-stub data-v-acb3ac67=\\"\\"></qfc-user-menu-stub>
  </nav>
</header>"
`;
